crates/pera-core/src/signature_mpc/mpc_manager.rs

```diff
@@ -232,49 +300,82 @@ impl SignatureMPCManager {
         Ok(())
     }
 
-    pub fn handle_mpc_message(
+    pub async fn handle_end_of_delivery(&mut self) -> PeraResult {
+        let txs_to_send: Vec<ConsensusTransaction> = self
+            .mpc_instances
+            .iter_mut()
+            .filter(|(_, instance)| {
+                instance.status == MPCSessionStatus::Active
+                    && instance.pending_messages.len() >= self.mpc_threshold_number_of_parties
```

Comment:
This should be changed to something like `instance.can_advance()` which returns `True` if it has received enough
messages, whatever the definition of "enough" is. Because right now this will break if we forget to change it after
changing to a voting power-based threshold.